src/ConnectedMode.UnitTests/UI/ProjectSelection/ProjectSelectionViewModelTests.cs

```diff
@@ -277,6 +278,27 @@ public async Task AdapterGetAllProjectsAsync_ReturnsResponseFromAdapter(bool exp
         response.ResponseData.Should().BeEquivalentTo(expectedServerProjects);
     }
 
+    [TestMethod]
+    [DataRow(null)]
+    [DataRow("")]
+    public void ProjectSearchTerm_SearchTermNullOrEmpty_ReturnsFalse(string searchTerm)
+    {
+        testSubject.ProjectSearchTerm = searchTerm;
+
+        testSubject.HasSearchTerm.Should().BeFalse();
+    }
+
+    [TestMethod]
+    [DataRow("abc")]
+    [DataRow(" ")]
+    [DataRow("\t")]
```

Comment:
It would be nice to Trim the search term and count those cases as empty

Reply:
The HasSearchTerm property is used to determine what text we should show when the list of projects is entered. If the user enters an empty string " " and there are no projects found for that, then we should show the message accordingly. So when " " is entered the text should say "no project was found for the entered search term" and not "No project exists"

We can trim the SearchTerm property before sending it to SlCore, but I am expecting that the search is optimized for this cases, so I don't think it is needed to do it for now.